request.go

```diff
@@ -61,9 +61,6 @@ func (r *Request) Encode(v interface{}) {
 		return
 	}
 	r.Header.Set("Content-Type", "application/json")
-	if r.ContentLength < 0 && cw.n < chunkThreshold {
```

Comment:
You no longer need the counting writer a few lines above, you can just use `r` directly

Reply:
With the most recent changes, calling `Encode` won't overwrite `Content-Length` and keep it at zero (note that a zero value is valid as a body length), a `-1` is what triggers chunked encoding. 

We should either set the content length for the json.Marshaler to `-1` or set an explicit length

Reply:
`Request.Write` is called under the hood which entirely takes care of setting ContentLength now, so why do we need to do anything special in the `json.Marshaler` case where we are not replacing `r.Body`?

Reply:
Yes, I didn't put the two together. `r` satisfies `io.Writer` and thus implicitly calls `Write` for the encoder.

Reply:
As discussed over DM, the json.Marshaler does indeed set content length (and tests verify this)